src/main/java/pl/agh/edu/master_diet/service/RecentSearchedService.java

```diff
@@ -0,0 +1,52 @@
+package pl.agh.edu.master_diet.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.agh.edu.master_diet.core.model.database.RecentProduct;
+import pl.agh.edu.master_diet.core.model.database.UserActivity;
+import pl.agh.edu.master_diet.core.model.rest.browser.activity.RecentActivityResponse;
+import pl.agh.edu.master_diet.core.model.rest.browser.product.RecentProductsResponse;
+import pl.agh.edu.master_diet.repository.RecentProductsRepository;
+import pl.agh.edu.master_diet.repository.UserActivityRepository;
+import pl.agh.edu.master_diet.service.converter.ConversionService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class RecentSearchedService {
+
+    private final ConversionService conversionService;
+    private final UserActivityRepository userActivityRepository;
+    private final PageFittingService<UserActivity> userActivityPageFittingService;
+    private final RecentProductsRepository recentProductsRepository;
+    private final PageFittingService<RecentProduct> recentProductPageFittingService;
+
+    public RecentActivityResponse getRecentActivities(Integer pageIndex, Integer perPage, Long userId) {
+        List<UserActivity> result = userActivityRepository.findByUserId(userId);
```

Comment:
I would move ```userActivityRepository.findByUserId(userId);``` to some new class ```UserActivityService``` and use optional   or else throw - maybe it will be useful in the future

Reply:
I thought about it too, but it will be too much fragmentation. Class with only one method isn't really good.

Reply:
Writing the same code multiple times isn't better, in fact, this might be reusable in the future once we use service

Reply:
I change it back to ActivityBrowserService and ProductBrowserService